tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/Continuations/ValueTaskContinuationGenerator.cs

```diff
@@ -2,170 +2,170 @@
-                    Resolver = new SyncCallbackHandler(continuation, result.PreserveContext);
-                }
+                var asyncContinuation = (AsyncObjectContinuationMethodDelegate)result.Method.CreateDelegate(typeof(AsyncObjectContinuationMethodDelegate));
+                Resolver = new AsyncCallbackHandler(asyncContinuation, result.PreserveContext);
             }
             else
             {
-                Resolver = new NoOpCallbackHandler();
-            }
-
-            if (Log.IsEnabled(LogEventLevel.Debug))
-            {
-                Log.Debug(
-                    "== {TaskContinuationGenerator} using Resolver: {Resolver}",
-                    $"TaskContinuationGenerator<{typeof(TIntegration).FullName}, {typeof(TTarget).FullName}, {typeof(TReturn).FullName}>",
-                    Resolver.GetType().FullName);
+                var continuation = (ObjectContinuationMethodDelegate)result.Method.CreateDelegate(typeof(ObjectContinuationMethodDelegate));
+                Resolver = new SyncCallbackHandler(continuation, result.PreserveContext);
             }
         }
+        else
+        {
+            Resolver = new NoOpCallbackHandler();
+        }
 
-        public override TReturn SetContinuation(TTarget instance, TReturn returnValue, Exception exception, in CallTargetState state)
+        if (Log.IsEnabled(LogEventLevel.Debug))
         {
-            return Resolver.ExecuteCallback(instance, returnValue, exception, in state);
+            Log.Debug(
+                "== {TaskContinuationGenerator} using Resolver: {Resolver}",
+                $"TaskContinuationGenerator<{typeof(TIntegration).FullName}, {typeof(TTarget).FullName}, {typeof(TReturn).FullName}>",
+                Resolver.GetType().FullName);
         }
+    }
+
+    public override TReturn? SetContinuation(TTarget? instance, TReturn? returnValue, Exception? exception, in CallTargetState state)
+    {
+        return Resolver.ExecuteCallback(instance, returnValue, exception, in state);
+    }
 
-        private class SyncCallbackHandler : CallbackHandler
+    private class SyncCallbackHandler : CallbackHandler
+    {
+        private readonly ObjectContinuationMethodDelegate _continuation;
+        private readonly bool _preserveContext;
+
+        public SyncCallbackHandler(ObjectContinuationMethodDelegate continuation, bool preserveContext)
         {
-            private readonly ObjectContinuationMethodDelegate _continuation;
-            private readonly bool _preserveContext;
+            _continuation = continuation;
+            _preserveContext = preserveContext;
+        }
 
-            public SyncCallbackHandler(ObjectContinuationMethodDelegate continuation, bool preserveContext)
+        public override TReturn? ExecuteCallback(TTarget? instance, TReturn? returnValue, Exception? exception, in CallTargetState state)
+        {
+            if (exception != null || returnValue == null)
```

Comment:
Are we bailing out here because `returnValue` can't _actually_ ever be null, because in this case `TReturn` must be a `ValueTask`? 🤔